mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageConverter.java

```diff
@@ -44,59 +45,66 @@
 import com.github.steveash.guavate.Guavate;
 import com.google.common.base.Preconditions;
 
-public class DeletedMessageConverter {
+class DeletedMessageConverter {
 
     private final Clock clock;
 
     public DeletedMessageConverter(Clock clock) {
         this.clock = clock;
     }
 
-    DeletedMessage convert(DeletedMessageMetadata deletedMessageMetadata, MailboxMessage mailboxMessage) throws IOException {
+    DeletedMessage convert(DeletedMessageMetadata deletedMessageMetadata, MailboxMessage mailboxMessage) {
         Preconditions.checkNotNull(deletedMessageMetadata);
         Preconditions.checkNotNull(mailboxMessage);
 
-        Message mimeMessage = parseMessage(mailboxMessage);
+        Optional<Message> mimeMessage = parseMessage(mailboxMessage);
 
         return DeletedMessage.builder()
             .messageId(deletedMessageMetadata.getMessageId())
             .originMailboxes(deletedMessageMetadata.getOwnerMailboxes())
             .user(retrieveOwner(deletedMessageMetadata))
-            .deliveryDate(retrieveDeliveryDate(mimeMessage))
+            .deliveryDate(retrieveDeliveryDate(mimeMessage, mailboxMessage))
             .deletionDate(ZonedDateTime.ofInstant(clock.instant(), ZoneOffset.UTC))
             .sender(retrieveSender(mimeMessage))
-            .recipients(retrieveRecipients(mimeMessage.getTo(), mimeMessage.getCc(), mimeMessage.getBcc()))
-            .hasAttachment(!mailboxMessage.getAttachments().isEmpty())
-            .subject(retrieveSubject(mimeMessage))
+            .recipients(retrieveRecipients(mimeMessage))
+            .hasAttachment(mailboxMessage.getAttachments().iterator().hasNext())
+            .subject(mimeMessage.map(Message::getSubject))
             .build();
     }
 
-    private Message parseMessage(MailboxMessage mailboxMessage) throws IOException {
+    private Optional<Message> parseMessage(MailboxMessage mailboxMessage) {
         DefaultMessageBuilder messageBuilder = new DefaultMessageBuilder();
         messageBuilder.setMimeEntityConfig(MimeConfig.PERMISSIVE);
         messageBuilder.setDecodeMonitor(DecodeMonitor.SILENT);
-        return messageBuilder.parseMessage(mailboxMessage.getFullContent());
+        try {
+            return Optional.ofNullable(messageBuilder.parseMessage(mailboxMessage.getFullContent()));
+        } catch (IOException e) {
```

Comment:
I'm not sure, at all, that it's what we want at the functional level

Reply:
Me I'm sure that failing deleting a message because it's not parsable is OK

IMO having "a few" missing fields in the vault is less bad

Reply:
There's two things here:
1. we could fail because message is not parseable (less likely, it was in a mailbox until now)
2. we could have a transient failure (IOException) when loading the content
In the second case, it really makes sense to fail the deletion.

Reply:
I catched the MimeIOException to catch the parse fail. And throw when IOException.